presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -375,11 +377,7 @@ private static void verifyEligibleToCompact(List<Block> blocks)
     {
         int[] newIds = new int[positionCount];
         for (int i = 0; i < positionCount; i++) {
-            int newId = remapIndex[getIndex(ids, i)];
-            if (newId == -1) {
```

Comment:
Because I know that every position is mapped and I suspected the test was a result of the faulty implementation.